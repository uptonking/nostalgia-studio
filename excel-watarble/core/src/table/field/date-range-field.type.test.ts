import { addMilliseconds } from 'date-fns';
import { type SafeParseError } from 'zod';
import {
  dateRangeFieldValue,
  type IDateRangeFieldValue,
} from './date-range-field.type';

const now = new Date();

test.each<[IDateRangeFieldValue, boolean]>([
  [[now, addMilliseconds(now, 1)], false],
  [[now, addMilliseconds(now, -1)], true],
  [[now, now], true],
  [null, false],
])('dateRangeFieldValue', (data, expectError) => {
  const result = dateRangeFieldValue.safeParse(data);

  if (!expectError) {
    expect(result.success).toBe(true);
  } else {
    expect(
      (result as SafeParseError<IDateRangeFieldValue>).error,
    ).toMatchSnapshot();
  }
});
