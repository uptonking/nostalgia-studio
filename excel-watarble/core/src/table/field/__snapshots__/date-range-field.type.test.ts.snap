// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`dateRangeFieldValue 1`] = `
[ZodError: [
  {
    "code": "custom",
    "message": "date range value from should before value to",
    "path": []
  }
]]
`;

exports[`dateRangeFieldValue 2`] = `
[ZodError: [
  {
    "code": "custom",
    "message": "date range value from should before value to",
    "path": []
  }
]]
`;
